### ISSUE TITLE
Removing files causes binary files to recompile

### ISSUE THREAD

**Issue by blinry** [nanoc/nanoc#1471]
### Steps to reproduce

Follow the steps in #1470. Then:

    $ cp ~/tmp/secondfile.png content
    $ nanoc
    $ rm content/secondfile.png
    $ nanoc show-data

### Expected behavior

We have removed a binary file. This should not cause another binary file to be invalidated.

### Actual behavior

It is!

### Details

This is the second problem I have in large projects. Adding new images is a fast operation, removing them is recompile everything hell! :fire:

**Comment by ddfreyne** [comment]
Can you run with the `NANOC_FEATURES=all` environment variable set, and try again?

This will enable the binary compiled content cache (which is behind a feature flag for now).

**Comment by blinry** [comment]
Oh, interesting! I tried that, and it didn't change the behavior for me. After removing the second image, `nanoc show-data` says that the first image depends on "(nothing)", but it's outdated because "The code snippets have been modified since the last time the site was compiled".

Let me know if I can provide a clearer way to reproduce this. :)

**Comment by ddfreyne** [comment]
> The code snippets have been modified since the last time the site was compiled

Whoa, odd — are you 100% sure that nothing is modifying anything in `lib/`?

**Comment by blinry** [comment]
I'm 99.5% sure! Can you reproduce that behavior with version 4.11.14?

**Comment by ddfreyne** [comment]
I’m having trouble reproducing this unfortunately. If you can create a Nanoc example site that exhibits the problem, that’d be helpful!

**Comment by blinry** [comment]
Sure! Tried to make it as minimal as possible: https://github.com/blinry/nanoc-issue-1471

I'm using nanoc 4.11.14 and Ruby 2.7.0. Steps to reproduce:

```
$ export NANOC_FEATURES=all
$ nanoc
$ rm content/image2.jpg
$ nanoc show-data
...
item /image1.jpg, rep thumbnail:
  is outdated:
    - The code snippets have been modified since the last time the site was compiled.
...
```

My expectation would be that image1.jpg would not be outdated after removing image2.jpg. Let me know if I can help or clarify more!

**Comment by ddfreyne** [comment]
Excellent job on the minimal reproduction case — I figured out the problem, and a fix is ready in #1493.

**Comment by blinry** [comment]
Can confirm! Brilliant, thank you so much, this removes a huge annoyance! <3
